client/dashboard/store-performance/index.js

```diff
@@ -35,9 +36,16 @@ import './style.scss';
 class StorePerformance extends Component {
 	constructor( props ) {
 		super( props );
+		const { title } = props;
+
 		this.state = {
 			hiddenIndicators: props.hiddenIndicators || [],
+			titleInput: title || __( 'Store Performance', 'woocommerce-admin' ),
 		};
+
+		this.onMenuToggle = this.onMenuToggle.bind( this );
+		this.onTitleChange = this.onTitleChange.bind( this );
+		this.onTitleBlur = this.onTitleBlur.bind( this );
```

Comment:
What do you think about moving these three components to `<CustomizableDashboard />`?

This way its less code to write when adding Sections (for extensions as well as us) and `onMenuToggle` can read the title from state when `titleInput === ''`. Plus 2 new methods (remove and moveUp/moveDown) will also follow the same pattern so it would be nice to have all in the same place.

Reply:
Agree with that, there is a lot of duplicated code there. Instead of moving this into `<CustomizableDashboard />` I created a new component named `<Section />` and moved those events and the `titleInput` state there (e2c339a3a6e27d69c379353eec9a9256c01371c0). The reason to create a new component instead of moving everything to `<CustomizableDashboard />` is that I think the input state doesn't need to be sent up to the dashboard on every key stroke and it's better to have it contained in the section. So the Dashboard only gets updated when the input is unfocused.

I imagine it's going to be different for the moveUp/moveDown buttons, since they don't need an internal state.

Let me know how this solution looks!